lib/loader/index.js

```diff
@@ -83,51 +87,39 @@ function preInstantiate(imports) {
   return extendedExports;
 }
 
-const E_NOEXPORTRUNTIME = "Operation requires compiling with --exportRuntime";
-const F_NOEXPORTRUNTIME = function() { throw Error(E_NOEXPORTRUNTIME); };
-
 /** Prepares the final module once instantiation is complete. */
 function postInstantiate(extendedExports, instance) {
   const exports = instance.exports;
   const memory = exports.memory;
   const table = exports.table;
-  const __new = exports.__new || F_NOEXPORTRUNTIME;
-  const __pin = exports.__pin || F_NOEXPORTRUNTIME;
-  const __unpin = exports.__unpin || F_NOEXPORTRUNTIME;
-  const __collect = exports.__collect || F_NOEXPORTRUNTIME;
+  const __new = exports.__new || F_NO_EXPORT_RUNTIME;
+  const __pin = exports.__pin || F_NO_EXPORT_RUNTIME;
+  const __unpin = exports.__unpin || F_NO_EXPORT_RUNTIME;
+  const __collect = exports.__collect || F_NO_EXPORT_RUNTIME;
   const __rtti_base = exports.__rtti_base;
-  const getRttiCount = __rtti_base
+  const getRttCount = __rtti_base
     ? function (arr) { return arr[__rtti_base >>> 2]; }
-    : F_NOEXPORTRUNTIME;
+    : F_NO_EXPORT_RUNTIME;
 
   extendedExports.__new = __new;
   extendedExports.__pin = __pin;
   extendedExports.__unpin = __unpin;
   extendedExports.__collect = __collect;
 
-  /** Gets the runtime type info for the given id. */
-  function getInfo(id) {
+  /** Gets the runtime type info for the given id and specific field. */
+  function getRtt(id, field = 0) {
     const U32 = new Uint32Array(memory.buffer);
-    const count = getRttiCount(U32);
-    if ((id >>>= 0) >= count) throw Error(`invalid id: ${id}`);
-    return U32[(__rtti_base + 4 >>> 2) + id * 2];
+    if ((id >>>= 0) >= getRttCount(U32)) throw Error(`invalid id: ${id}`);
+    return U32[(__rtti_base + 4 >>> 2) + (id << 1) + field];
```

Comment:
Seems a little brittle, in that it only works right now because existing fields are 32-bit. Perhaps a helper to compute the *byte* offset, that is then re-used by `getRtti` and `getBase`?

Reply:
Perhaps better names could be `getRttInfo` and `getRttBase`?

Reply:
Yeah, I'll just revert this changes